### ISSUE TITLE
Enourmous numbers of roi proposals with Defaultpredictor() even model training seems fine

### ISSUE THREAD

**Issue by colt18** [facebookresearch/detectron2#2034]
Problem definition:
Getting extremely large numbers of false roi proposals.

Case:
I do have a database which consists of 800 1000x1000 resolution grayscale images. Well annotated and converted into coco json. Registered and dicts are formed as below:

```
register_coco_instances("ds", {}, r"train/ds.json", r"train/input")

dataset_dicts = detectron2.data.DatasetCatalog.get("ds")
metadata_dicts = detectron2.data.MetadataCatalog.get("ds")
```

I do a sanity check by visualizing images if they are proper and it seems ok. Bounding boxes are well placed for all images. Just an example of one.

```
INPUT_DIR = r"train/input"
file = "train00001.png"
img = cv2.imread(os.path.join(INPUT_DIR,file))
visualizer = Visualizer(img)
vis = visualizer.draw_dataset_dict(dataset_dicts[0])
cv2.imshow("img",vis.get_image())
```

![we](https://user-images.githubusercontent.com/26602153/93254467-0c841700-f7a1-11ea-8c24-1599d6615a0d.png)

Setting up the config and the training like this fashion.  Using same dataset for testing purposes just ignore it that is intentional::

```
config_file = "COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml"
cfg = get_cfg()
cfg.merge_from_file(model_zoo.get_config_file(config_file))
cfg.DATASETS.TRAIN = ('ds',)
cfg.DATASETS.TEST = ('ds',)

cfg.DATALOADER.NUM_WORKERS = 4

cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url(config_file)  

cfg.SOLVER.IMS_PER_BATCH = 4
cfg.SOLVER.BASE_LR = 0.001
cfg.SOLVER.WARMUP_ITERS = 1000
cfg.SOLVER.MAX_ITER = 30000
cfg.SOLVER.STEPS = (1000, 1500)
cfg.SOLVER.GAMMA = 0.05

cfg.MODEL.ROI_HEADS.NUM_CLASSES = 3

cfg.TEST.EVAL_PERIOD = 500

from detectron2.engine import DefaultTrainer
os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
trainer = DefaultTrainer(cfg) 
trainer.resume_or_load(resume=True)
trainer.train()
```
Training seems fine. Here are some metrics after somewhile:

`[09/15 21:30:22 d2.utils.events]:  eta: 1:23:01  iter: 25919  total_loss: 0.9085  loss_cls: 0.1881  loss_box_reg: 0.2199  loss_mask: 0.3752  loss_rpn_cls: 0.03945  loss_rpn_loc: 0.1211  time: 0.6981  data_time: 0.0018  lr: 2.5e-06  max_mem: 3847M`

Now predicting with DefaultPredictor and visualizing the masks. I'm using the training examples:

```
predictor = DefaultPredictor(cfg)
INPUT_DIR = r"train/input"
file = "train00001.png"
img = cv2.imread(os.path.join(INPUT_DIR,file))
visualizer = Visualizer(img)
vis = visualizer.draw_dataset_dict(dataset_dicts[0])
cv2.imshow("img",vis.get_image())
```

And this is the result. There are 3vehicles in this image that one can be spotted at the bottom purple region and two are lost under the masks and none detected. So what I endup with is useless random rois. There are actually a total of 100 rois out there acording to `instances.scores` :

![ww](https://user-images.githubusercontent.com/26602153/93253503-a0ed7a00-f79f-11ea-8117-9a2d6bcf627c.png)

**Comment by ppwwyyxx** [comment]
As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.

We provide configs & models with standard academic settings and expect users to have the knowledge to choose or design appropriate models & parameters for their own tasks.

**Comment by ppwwyyxx** [comment]
> Now predicting with DefaultPredictor and visualizing the masks. I'm using the training examples:
predictor = DefaultPredictor(cfg)
INPUT_DIR = r"train/input"
file = "train00001.png"
img = cv2.imread(os.path.join(INPUT_DIR,file))
visualizer = Visualizer(img)
vis = visualizer.draw_dataset_dict(dataset_dicts[0])
cv2.imshow("img",vis.get_image())

This code is visualizing annotations but not predictions. This doesn't seem what you expect to do
